app/controllers/concerns/decidim/proposals/admin/valuation_assignments_controller_override.rb

```diff
@@ -23,6 +23,28 @@ def create
             end
           end
 
+          def destroy
+            @form = form(Admin::ValuationAssignmentForm).from_params(destroy_params)
+
+            enforce_permission_to :unassign_from_valuator, :proposals, valuator: @form.valuator_user
+
+            Admin::UnassignProposalsFromValuator.call(@form) do
+              on(:ok) do |_proposal|
+                flash.keep[:notice] = I18n.t("valuation_assignments.delete.success", scope: "decidim.proposals.admin")
+                if current_user == @form.valuator_user
+                  redirect_to EngineRouter.admin_proxy(current_component).root_path
+                else
+                  redirect_back fallback_location: EngineRouter.admin_proxy(current_component).root_path
+                end
```

Comment:
I understand this override is just to redirect the user to different places depending on if it is a valuator or not.

Can you explain why we need that, also we should add a test case for both cases if it is really needed